server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -159,7 +160,7 @@ public RepositoriesService(
      * @param request  register repository request
      * @param listener register repository listener
      */
-    public void registerRepository(final PutRepositoryRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
+    public void registerOrUpdateRepository(final PutRepositoryRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
```

Comment:
When we try to register a new system repository, do we call this method? or that flow would be separate?

Reply:
That should be a separate flow